pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/retention/RetentionManager.java

```diff
@@ -271,4 +276,15 @@ private void manageSegmentLineageCleanupForTable(String tableNameWithType) {
     }
     LOGGER.info("Segment lineage metadata clean-up is successfully processed for table: {}", tableNameWithType);
   }
+
+  private boolean shouldDeleteReplacedSegments(TableConfig tableConfig, LineageEntry lineageEntry) {
```

Comment:
It'd be good to add a javadoc for this method to elaborate that the following cases are qualified to be deleted:
* The table is an APPEND use case
*  The segment lineage has been completed for 24 hrs.